hawkbit-autoconfigure/src/main/java/org/eclipse/hawkbit/autoconfigure/security/InMemoryUserManagementAutoConfiguration.java

```diff
@@ -11,129 +11,155 @@
-    UserDetailsService userDetailsService() {
-
-        final List<UserPrincipal> userPrincipals = new ArrayList<>();
-        for (final MultiUserProperties.User user : multiUserProperties.getUsers()) {
-            final List<String> permissions = user.getPermissions();
-            List<GrantedAuthority> authorityList;
-            // Allows ALL as a shorthand for all permissions
-            if (permissions.size() == 1 && "ALL".equals(permissions.get(0))) {
-                authorityList = PermissionUtils.createAllAuthorityList();
-            } else {
-                authorityList = createAuthoritiesFromList(permissions);
-            }
-
-            final UserPrincipal userPrincipal = new UserPrincipal(user.getUsername(), user.getPassword(),
-                    user.getFirstname(), user.getLastname(), user.getUsername(), user.getEmail(), DEFAULT_TENANT,
-                    authorityList);
+    private static UserDetailsService userDetailsService(
+            final SecurityProperties securityProperties,
+            final TenantAwareUserProperties tenantAwareUserProperties,
+            final PasswordEncoder passwordEncoder) {
+        final List<User> userPrincipals = new ArrayList<>();
+        tenantAwareUserProperties.getUsers().forEach((username, user) -> {
+            final TenantAwareUser userPrincipal = new TenantAwareUser(
+                    username, password(user.getPassword(), passwordEncoder),
+                    createAuthorities(user.getRoles(), Collections::emptyList),
+                    DEFAULT_TENANT);
             userPrincipals.add(userPrincipal);
-        }
+        });
 
-        // If no users are configured through the multi user properties, set up
-        // the default user from security properties
+        // If no tenant users are configured through the tenant user properties, set up
+        // the default user from spring security properties as super DEFAULT tenant user
         if (userPrincipals.isEmpty()) {
-            final String name = securityProperties.getUser().getName();
-            final String password = securityProperties.getUser().getPassword();
-            final List<String> roles = securityProperties.getUser().getRoles();
-            final List<GrantedAuthority> authorityList = roles.isEmpty() ? PermissionUtils.createAllAuthorityList()
-                    : createAuthoritiesFromList(roles);
             userPrincipals
-                    .add(new UserPrincipal(name, password, name, name, name, null, DEFAULT_TENANT, authorityList));
+                    .add(new TenantAwareUser(
+                            securityProperties.getUser().getName(),
+                            password(securityProperties.getUser().getPassword(), passwordEncoder),
+                            createAuthorities(
+                                    securityProperties.getUser().getRoles(), PermissionUtils::createAllAuthorityList),
+                            DEFAULT_TENANT));
```

Comment:
here we always put DEFAULT_TENANT ?

Reply:
yes. in case there are "tenant users" we treat spring user as "global / non-tenant user" - such are already defined
if there are no "tenant users" - we that it as default tenant user for default tenant DEFAULT